asv_bench/benchmarks/merge.py

```diff
@@ -13,6 +15,49 @@ def setup(self, existing_elements):
             d[f"var{i}"] = i
         self.dataset = xr.merge([d])
 
+        d = {"set_2_{i}": i for i in range(existing_elements)}
+        self.dataset2 = xr.merge([d])
+
     def time_variable_insertion(self, existing_elements):
         dataset = self.dataset
         dataset["new_var"] = 0
+
+    def time_merge_two_datasets(self, existing_elements):
+        xr.merge([self.dataset, self.dataset2])
+
+
+class DatasetCreation:
+    param_names = ["strategy", "count"]
+    params = [
+        ["dict_of_DataArrays", "dict_of_Variables", "dict_of_Tuples"],
+        [0, 1, 10, 100, 1000],
+    ]
+
+    def setup(self, strategy, count):
+        self.dataset_value = np.array(["0", "b"], dtype=str)
+        self.dataset_coords = dict(time=np.array([0, 1]))
+
+    def time_dataset_creation(self, strategy, count):
+        # The idea here is to time how long it takes to go from numpy
+        # and python data types, to a full dataset
+        # See discussion
+        # https://github.com/pydata/xarray/issues/7224#issuecomment-1292216344
+        if strategy == "dict_of_DataArrays":
```

Comment:
Can we move these if-checks to the setup part?

Reply:
you want to move the whole DataArray variable creation there as well? I'm confused what the benchmark would look like.

Reply:
I was thinking wrapping it in a function in `setup`, something like this:
```python
def create_data_vars():
    return {
        f"long_variable_name_{i}": xr.Variable("time", self.dataset_value)
        for i in range(count)
    }
self.create_data_vars  = create_data_vars
```
then call `self.create_data_vars()` in `time_dataset_creation`.

Reply:
thnaks.